src/main/java/org/folio/circulation/resources/DeclareLostResource.java

```diff
@@ -0,0 +1,69 @@
+package org.folio.circulation.resources;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.folio.circulation.support.Result.failed;
+import static org.folio.circulation.support.Result.succeeded;
+import static org.folio.circulation.support.ValidationErrorFailure.singleValidationError;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+
+import io.vertx.core.http.HttpClient;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import org.folio.circulation.domain.Loan;
+import org.folio.circulation.domain.LoanRepository;
+import org.folio.circulation.domain.representations.DeclareItemLostRequest;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.NoContentResult;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.http.server.WebContext;
+
+public class DeclareLostResource extends Resource {
+
+  public DeclareLostResource(HttpClient client) {
+    super(client);
+  }
+
+  @Override
+  public void register(Router router) {
+    router.put("/circulation/loans/:id/declare-item-lost")
+      .handler(this::declareLost);
+  }
+
+  private void declareLost(RoutingContext routingContext) {
+    final WebContext context = new WebContext(routingContext);
+    final Clients clients = Clients.create(context, client);
+    final LoanRepository loanRepository = new LoanRepository(clients);
+
+    validateDeclaredLostRequest(routingContext).thenCompose(r -> r.after(request ->
```

Comment:
Personally, I’d put the `thenCompose` invocation on the following line, to make the chain clearer for readers.

Also, it might reveal an unnecessary nesting, why is the closed validation and process itself nested deeper than the others?

Reply:
This has been changed. Note that the `request` param is used in 2 following invocations.